clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/caching/providers/TitusTargetGroupServerGroupProvider.java

```diff
@@ -133,4 +125,75 @@ public TitusTargetGroupServerGroupProvider(Cache cacheView) {
     return source.getRelationships().get(relationship) != null ? cacheView.getAll(relationship, source.getRelationships().get(relationship)) : Collections.emptyList();
   }
 
+  private Optional<LoadBalancerInstance> getInstanceHealth(String instanceKey,
+                                                          Map instanceDetails,
+                                                          Map<String, String> targetGroupDetails) {
+    String healthKey;
+    try {
+      healthKey = com.netflix.spinnaker.clouddriver.aws.data.Keys.getInstanceHealthKey(
+        ((Map) instanceDetails.get("task")).get("containerIp").toString(),
+        targetGroupDetails.get("account"),
+        targetGroupDetails.get("region"),
+        "aws-load-balancer-v2-target-group-instance-health"
+      );
+    } catch (NullPointerException e) {
+      return Optional.empty();
+    }
+
+    CacheData healthData = cacheView.get(HEALTH.ns, healthKey);
+
+    Map<String, Object> health = getTargetGroupHealth(instanceKey, targetGroupDetails, healthData);
+
+    return Optional.of(new LoadBalancerInstance(
+      ((Map) instanceDetails.get("task")).get("id").toString(),
```

Comment:
I guess it's safe to say an instance will always have a task will always have an id ... we weren't guarding against this previously and I can't think of how this wouldn't hold true